### PR TITLE
Rework Drink/Food/FoodContainer entirely

### PR DESCRIPTION
**Pull request by FL-OZ**

### CODE REVIEW

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs [hunk 1]
```diff
@@ -1,148 +1,179 @@
...
+            {
+                Owner.Name = string.Empty;
+                Owner.Name = $"{_originalName} {Loc.GetString("(opened)")}";
+            }
+
+            if (Opened && Empty)
+            {
+                Owner.Name = string.Empty;
+                Owner.Name = $"{_originalName} {Loc.GetString("(empty)")}";
+            }
```
Comment (Clyybber) [inline Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:101]:
```suggestion
            if (Opened)
            {
                if (Empty)
                {
                    Owner.Name = string.Empty;
                    Owner.Name = $"{_originalName} {Loc.GetString("(empty)")}";
                }
                else
                {
                    Owner.Name = string.Empty;
                    Owner.Name = $"{_originalName} {Loc.GetString("(opened)")}";
                }
            }
```

Comment (FL-OZ) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:101]: :facepalm:

File: Content.Server/GameObjects/Components/Nutrition/FoodContainer.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    /// <summary>
+    /// This container acts as a master object for things like Pizza, which holds slices.
+    /// </summary>
+    /// TODO: Perhaps implement putting food back (pizza boxes) but that really isn't mandatory.
+    /// This doesn't even need to have an actual Container for right now.
+    [RegisterComponent]
+    public sealed class FoodContainer : SharedFoodContainerComponent, IUse
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/Nutrition/FoodContainer.cs:22]: Rename this class to FoodContainerComponent.

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs [hunk 3]
```diff
@@ -1,148 +1,179 @@
...
         }
 
-        protected override void Startup()
+        public override void Initialize()
         {
-            base.Startup();
-            _contents = Owner.GetComponent<SolutionComponent>();
+            base.Initialize();
+            Owner.TryGetComponent(out _appearanceComponent);
+            Owner.TryGetComponent<SolutionComponent>(out _contents);
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:69]: Why did you change to using `TryGet` instead of just `Get`?

Comment (FL-OZ) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:69]: oh, i did. wtf.

Comment (FL-OZ) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:69]: it looks better and doesnt crash the game if a drink bottle is doesnt have a solutioncomponent

Comment (ComicIronic) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:69]: But `_contents` is used in a non-null way.

Comment (PJB3005) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:69]: This should be fixed, yes.

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs [hunk 4]
```diff
@@ -1,148 +1,179 @@
...
 
-        bool IUse.UseEntity(UseEntityEventArgs eventArgs)
+        private void UpdateName()
         {
-            UseDrink(eventArgs.User);
+            if (Opened)
+            {
+                if (Empty)
+                {
+                    Owner.Name = $"{_originalName} {Loc.GetString("(empty)")}";
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:95]: Do we even need an `unopened` text like that? I would maybe be nicer if you just saw it on the sprite/examine and not change the name.

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs [hunk 5]
```diff
@@ -1,148 +1,179 @@
...
+                _audioSystem.Play(_useSound, Owner, AudioParams.Default.WithVolume(-2f));
+                target.PopupMessage(target, Loc.GetString("Slurp"));
+                UpdateName();
+                UpdateAppearance();
+                return true;
+            }
+
+            //Stomach was full or can't handle whatever solution we have.
+            _contents.TryAddSolution(split);
+            target.PopupMessage(target, Loc.GetString($"You've had enough {Owner.Name}!"));
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:175]: This should be `Loc.GetString("You've had enough {0}!", Owner.Name)` instead.

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs [hunk 6]
```diff
@@ -1,148 +1,179 @@
...
-            if (!InteractionChecks.InRangeUnobstructed(eventArgs)) return;
+            TryUseDrink(eventArgs.Target);
+        }
 
-            UseDrink(eventArgs.Target);
+        public void Examine(FormattedMessage message)
+        {
+            var color = Opened ? "yellow" : "white";
+            var openedText = Opened ? Loc.GetString("opened") : Loc.GetString("closed");
+            message.AddMarkup(Loc.GetString($"It is [color={color}]{openedText}[/color].") );
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:135]: This should be `Loc.GetString("It is [color={0}]{1}[/color].", color, openedText)` instead.

File: Content.Server/GameObjects/Components/Nutrition/FoodContainerComponent.cs [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IRobustRandom _random;
+        [Dependency] private readonly IEntityManager _entityManager;
+#pragma warning restore 649
+        public override string Name => "FoodContainer";
+
+        private AppearanceComponent _appearance;
+        private Dictionary<string, int> _prototypes;
+        private uint _capacity;
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Nutrition/FoodContainerComponent.cs:32]: I don't see the point of storing this as a uint.

Comment (FL-OZ) [reply Content.Server/GameObjects/Components/Nutrition/FoodContainerComponent.cs:32]: It will never be negative.

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs (hunk 6 above)
Comment (FL-OZ) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:135]: Thanks thats a lot better.

File: Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs (hunk 4 above)
Comment (FL-OZ) [reply Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs:95]: It's on both!

File: Content.Server/GameObjects/Components/Nutrition/FoodContainerComponent.cs (hunk 7 above)
Comment (boiled-water-tsar) [reply Content.Server/GameObjects/Components/Nutrition/FoodContainerComponent.cs:32]: those are some famous last words right there

Comment (PJB3005) [reply Content.Server/GameObjects/Components/Nutrition/FoodContainerComponent.cs:32]: Yeah but you're returning it as an int in public API so:
1. If you actually need that extra space, you have a bug
2. If you don't need it (you don't), you shouldn't be using a uint because it's annoying.

**Comment by FL-OZ** [comment]
merge it
